AutomatticTracks/src/main/java/com/automattic/android/tracks/crashlogging/CrashLogging.kt

```diff
@@ -0,0 +1,101 @@
+package com.automattic.android.tracks.crashlogging
+
+import android.content.Context
+import com.automattic.android.tracks.BuildConfig
+import com.automattic.android.tracks.crashlogging.internal.SentryErrorTrackerProxy
+import com.automattic.android.tracks.crashlogging.internal.SentryErrorTrackerProxyImpl
+import io.sentry.SentryEvent
+import io.sentry.SentryLevel
+import io.sentry.SentryOptions
+import io.sentry.protocol.Message
+import io.sentry.protocol.User
+
+object CrashLogging {
+
+    private lateinit var dataProvider: CrashLoggingDataProvider
+    private lateinit var sentryProxy: SentryErrorTrackerProxy
+
+    @JvmStatic
+    fun start(
+        context: Context,
+        dataProvider: CrashLoggingDataProvider,
+    ) {
+        start(context, dataProvider, SentryErrorTrackerProxyImpl())
+    }
+
+    internal fun start(
```

Comment:
This method allows us to test any logic for `CrashLogging`. I see no other option, as the static nature of `CrashLogging` won't allow us to inject an object `SentryErrorTrackerProxy` in any other way.

I know that production code shouldn't be bothered by tests. In this case though, I believe there are more advantages than harm: this method is `internal` (won't leak to library consumers) and tests written for this class assert, that all fields have been correctly assigned.

Do you have any opinion or suggestion on that?

Reply:
I wish CrashLogging wasn't an object - static classes/methods just make testing more difficult. I understand that refactoring it into an injectable class is above the scope of this PR. So with the current constraints, I think this approach is ok. 
One thing we might want to consider is using @VisibleForTesting annotation.  It documents the fact that the methods signature was modified for tests and the tools will complain if we use the method from production code. Wdyt?

Reply:
> I wish CrashLogging wasn't an object - static classes/methods just make testing more difficult.

Yes, that's true. I think migrating this to an object isn't difficult for a library, rather for Tracks consumers as we would have to start injecting this object in many places - I don't know if WPAndroid or WCAndroid (or SimpleNote) are ready for that... 😅 


> One thing we might want to consider is using @VisibleForTesting annotation.

Sure, added in 1f600e65cf2b8f779a812b17cf32f5c3b50c8c1a

Reply:
> Yes, that's true. I think migrating this to an object isn't difficult for a library, rather for Tracks consumers as we would have to start injecting this object in many places - I don't know if WPAndroid or WCAndroid or SimpleNote are ready for that... 😅

I don't know about WCAndroid and SimpleNote, but it should be a rather simple change for WPAndroid. I think other repos could create a static wrapper if they preferred the static access. So I don't think it'd be too much effort. Having said that it's not focus of this PR so lets not worry about it.

> Sure, added in 1f600e6

Thanks!

Reply:
> I don't know about WCAndroid and SimpleNote, but it should be a rather simple change for WPAndroid

That's sounds good! I'll keep that in mind with future work on Tracks